domain/consensus/processes/coinbasemanager/block_reward_switch.go

```diff
@@ -0,0 +1,49 @@
+package coinbasemanager
+
+import (
+	"github.com/kaspanet/kaspad/domain/consensus/model"
+	"math/big"
+)
+
+func (c *coinbaseManager) isBlockRewardFixed(stagingArea *model.StagingArea) (bool, error) {
+	if c.hasBlockRewardSwitchedToFixed {
+		return true, nil
+	}
+
+	currentPruningPointIndex, err := c.pruningStore.CurrentPruningPointIndex(c.databaseContext, stagingArea)
```

Comment:
You should check from the pruning point of the current checked block POV (the one in its header), otherwise you can encounter subjectivity issues.